ecl/hqlcpp/hqlsource.cpp

```diff
@@ -2581,6 +2656,26 @@ void SourceBuilder::gatherVirtualFields(bool ignoreVirtuals, bool ensureSerializ
         physicalRecord.setown(fieldInfo.createPhysicalRecord());
     else
         physicalRecord.set(record);
+
+    if (newDiskReadMapping)
+    {
+        projectedRecord.set(tableExpr->queryRecord());
+        expectedRecord.setown(getSerializedForm(physicalRecord, diskAtom));
+        //MORE: Reduce projected to the fields that are actually required by the dataset, and will need to remap field references.
```

Comment:
I take it this is the subject of a subsequent Jira? I wonder if worth referencing the Jira in the MORE comment in such cases?